### PR TITLE
feat(centos-6): reshape formula and tests for this platform

### LINKED ISSUE #104: ci(kitchen) service tests should verify a service common across platforms
On the initial incarnations of the tests, we added `systemd-udevd` as the service to check. Now that we added `centos-6` platform, which is `SysV`, we should probably modify the test example to verify some other service, perhaps `crond`

### PR DESCRIPTION
**Pull request by myii**
* Close #104

---

This is an example of how #104 could be resolved, using `crond` as suggested by @javierbertoli, while making use of the `*.yaml` files as suggested by @baby-gnu.  This allows discussions to begin about how to finalise this implementation.

### CODE REVIEW

**Comment by myii** [comment]
@daks Thanks for the review.

As a general note, I'm in two minds about leaving it with this "minimal intervention" or converting globally to `cron` instead of testing for `systemd`.

File: template/defaults.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # vim: ft=yaml
 ---
 template:
-  pkg: template
+  pkg: bash
```
Comment (javierbertoli) [inline template/defaults.yaml]: I think that the way it was before allows us to kept two clearly differentiated groups of files:

* the *formula-related* ones, which are all of those that have `template-something` or `something-template` or 'template' in the name.
* the *testing-related* ones, which are not intrinsically needed by the formulas: that's where (I think) we should put these examples, like `pkg: bash` or `service: whatever` 

I think the former group should be respected and kept free of 'testing-related' content, while the latter are the place for whatever we consider useful for testing (in fact, this is the same reason why I think that the linting and semver tools/files should be moved under `test`, at least).

If the content of the **formula-related** files is named `template` then I can simply go with a 'search & replace' for 's/template/my_package/g', the usage of a 'templated formula' is clear, and the entry-level for newcomers is easy (I usually do this, in fact :smile:).

However, if a newcomer needs to use the formula, 'search manually all around and make sense what to replace and what not, what is a requirement and what is an example', the template loses part of its value: 

'what should I replace? `pkg: bash` in the `template/defaults.yaml`? or that is a dependency that should not be modified?' 'same goes to `service: systemd-udev`?' 

'oh, I replaced `pkg: bash` but forgot of `pkg: cronie` in `template/osfingermap.yaml`. Is it a requirement or an example?'

**Comment by javierbertoli** [comment]
> As a general note, I'm in two minds about leaving it with this "minimal intervention" or converting globally to cron instead of testing for systemd.

Following what I mentioned above, I think that the reason we're in this situation is because we're providing 'a testing framework' to be used with the future 'template-derived' formula but, at the same time, using the same framework to test the 'template' itself. This mean we need to keep the testing template simple while, at the same time, it has to be useful for us :yum:

I thought (not much, really :smile: ) of using `cron` because it's generally named the same across different distros. Now, in hindsight, notice that I forgot of `cronie` and `vixie-cron`, etc.

Perhaps, to keep things simple, we can use some other package and service?

**Comment by myii** [comment]
@javierbertoli I can see benefits to both ways.  The points you make are definitely important.  On the flip side, it's nice having working examples of how the map runs the formula, i.e. how to specify the settings in the appropriate `.yaml` file.  If we do go with the way you've suggested, then there's a far simpler, albeit hackier, solution available.  We don't need to touch any of the `.yaml` files and just add a bit of Jinja to `pillar.example`:

```diff
+  {%- if grains.osfinger == 'CentOS-6' %}
+  pkg: cronie
+  service:
+    name: crond
+  {%- else %}
   pkg: bash
-  config: /etc/template-formula.conf
   service:
     name: systemd-udevd
+  {%- endif %}
+  config: /etc/template-formula.conf
```

And here's the working run for that:

* https://travis-ci.org/myii/template-formula/builds/530080602

This achieves the same thing as this PR.  How do you all feel about this method?

**Comment by myii** [comment]
@javierbertoli As discussed on Slack, using the Jinja as mentioned in the comment above.  So this is ready to be merged.

**Comment by javierbertoli** [comment]
Thanks @myii !

**Comment by myii** [comment]
You're welcome, thanks for the merge, @javierbertoli.

**Pull request merged**